File: Array/SingleNumberLeetcode/Vamshi-single-number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
 var singleNumber = function(nums) {
   var obj = {}
   for(var i of nums)
   {
     if(obj[i])
     {
       obj[i]=false
     }
     else
     {
       obj[i]=true
     }
   }
   for([key,value] of Object.entries(obj))
   {
     if(value)
     {
       return key
     }
   }
};
console.log(singleNumber([1,2,2]))

File: Array/SingleNumberLeetcode/Vamshi-single-number.py
from typing import List
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        obj = dict()
        for i in nums:
            if i in obj:
                obj[i]=False
            else:
                obj[i]=True
        for key,value in obj.items():
            if value==True:
                return key
        